packages/plugin-webpack/src/plugin.ts

```diff
@@ -0,0 +1,32 @@
+import { postProcess, PostProcessOptions } from '@metamask/snap-utils';
+import { Compiler, WebpackPluginInstance } from 'webpack';
+import { RawSource } from 'webpack-sources';
+
+const PLUGIN_NAME = 'SnapWebpackPlugin';
+
+export type Options = PostProcessOptions;
+
+export default class SnapWebpackPlugin implements WebpackPluginInstance {
+  public readonly options: Partial<Options>;
+
+  constructor(options: Partial<Options> = {}) {
+    this.options = options;
+  }
+
+  apply(compiler: Compiler) {
+    compiler.hooks.compilation.tap(PLUGIN_NAME, (compilation) => {
+      compilation.hooks.processAssets.tap(PLUGIN_NAME, (assets) => {
+        Object.keys(assets).forEach((assetName) => {
```

Comment:
Are the assets guaranteed to be JavaScript files here?

Reply:
Good question! I'll add a test case which processes other files, and update this code if necessary.